SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/ArrayTests.java

```diff
@@ -44,20 +58,238 @@ public void testArray() {
+        DBSPCircuit circuit = getCircuit(compiler);
+
+        DBSPZSetLiteral result = null;
+        for (int i = 1; i < 5; i++) {
+            DBSPTupleExpression tuple = new DBSPTupleExpression(new DBSPI32Literal(i, true));
+            if (i == 1)
+                result = new DBSPZSetLiteral(tuple);
+            else
+                Objects.requireNonNull(result).add(tuple);
+        }
+        result.add(new DBSPTupleExpression(DBSPLiteral.none(DBSPTypeInteger.NULLABLE_SIGNED_32)));
+
+        this.addRustTestCase(circuit, new InputOutputPair(
+                new DBSPZSetLiteral[0], new DBSPZSetLiteral[]{ result }));
+    }
+
+    @Test
+    public void testUnnestOrdinality() {
+        DBSPCompiler compiler = testCompiler();
+        String query = "CREATE VIEW V AS SELECT * FROM UNNEST(ARRAY [1, 2, 3, 4, 5]) WITH ORDINALITY";
+        compiler.compileStatements(query);
+        if (compiler.hasErrors())
+            compiler.showErrors(System.err);
+        DBSPCircuit circuit = getCircuit(compiler);
+
+        DBSPZSetLiteral result = null;
+        for (int i = 1; i < 6; i++) {
+            DBSPTupleExpression tuple = new DBSPTupleExpression(
+                    new DBSPI32Literal(i),
+                    new DBSPI32Literal(i));
+            if (i == 1)
+                result = new DBSPZSetLiteral(tuple);
+            else
+                Objects.requireNonNull(result).add(tuple);
+        }
+
+        this.addRustTestCase(circuit, new InputOutputPair(
+                new DBSPZSetLiteral[0], new DBSPZSetLiteral[]{ result }));
+    }
+
+    @Test
+    public void testUnnestOrdinalityNull() {
+        DBSPCompiler compiler = testCompiler();
+        String query = "CREATE VIEW V AS SELECT * FROM UNNEST(ARRAY [1, 2, 3, 4, 5, NULL]) WITH ORDINALITY";
+        compiler.compileStatements(query);
+        if (compiler.hasErrors())
+            compiler.showErrors(System.err);
+        DBSPCircuit circuit = getCircuit(compiler);
+
+        DBSPZSetLiteral result = null;
+        for (int i = 1; i < 6; i++) {
+            DBSPTupleExpression tuple = new DBSPTupleExpression(
+                    new DBSPI32Literal(i, true),
+                    new DBSPI32Literal(i));
+            if (i == 1)
+                result = new DBSPZSetLiteral(tuple);
+            else
+                Objects.requireNonNull(result).add(tuple);
+        }
+        result.add(new DBSPTupleExpression(
+                DBSPLiteral.none(DBSPTypeInteger.NULLABLE_SIGNED_32),
+                new DBSPI32Literal(6)));
+
+        this.addRustTestCase(circuit, new InputOutputPair(
+                new DBSPZSetLiteral[0], new DBSPZSetLiteral[]{ result }));
+    }
+
+    //@Test
+    public void testUnnest2() {
+        // TODO: the Calcite type seems to be wrong, it has no nullable columns
+        DBSPCompiler compiler = testCompiler();
+        String query = "CREATE VIEW V AS SELECT * FROM UNNEST(ARRAY [1, 2, 3, 4, 5], ARRAY[3, 2, 1])";
```

Comment:
All the tests seem to use integer arrays. Can there be TEXT arrays too?

Reply:
There will be more tests from Postgres. This just contains very simple tests.